src/client/datascience/interactive-ipynb/randomBytes.ts

```diff
@@ -0,0 +1,85 @@
+import { inject, injectable } from 'inversify';
+import { traceError, traceInfo } from '../../common/logger';
+import { IFileSystem, IPlatformService } from '../../common/platform/types';
+import { IProcessServiceFactory } from '../../common/process/types';
+import { OSType } from '../../common/utils/platform';
+import { ISystemPseudoRandomNumberGenerator } from '../types';
+
+// Wraps operating system-provided pseudorandom number generator facilities to provide
+// cryptographically secure random bytes.
+@injectable()
+export class SystemPseudoRandomNumberGenerator implements ISystemPseudoRandomNumberGenerator {
+    constructor(
+        @inject(IPlatformService) private readonly platformService: IPlatformService,
+        @inject(IProcessServiceFactory) private readonly processServiceFactory: IProcessServiceFactory,
+        @inject(IFileSystem) private readonly fileSystem: IFileSystem
+    ) {}
+
+    public async randomBytes(numBytes: number) {
+        switch (this.platformService.osType) {
+            case OSType.Windows:
+                return this.randomBytesForWindows(numBytes);
+            case OSType.Linux:
+            case OSType.OSX:
+                return this.randomBytesForUnixLikeSystems(numBytes);
+            default:
+                throw new Error('Cannot generate random bytes for unknown operating system.');
+        }
+    }
+
+    // Calls into BCryptGenRandom in bcrypt.dll using node-ffi
+    // See https://github.com/node-ffi/node-ffi/wiki/Node-FFI-Tutorial for details on usage
+    private randomBytesForWindows(numBytes: number) {
+        try {
+            // tslint:disable: no-require-imports
+            // Lazy-load modules required for calling BCryptGenRandom
+            const ffi = require('ffi-napi');
```

Comment:
```suggestion
            const ffi = require('ffi-napi') as typeof import('ffi-napi);
```

This way you get strong typing.
If you don't need then then we should uninstall the @types/ffi-napi & other type definitinos.